115 report

Mary249928 opened the pull request:
The complete report #115 
Sub issues done: #116 #117 #118

Gartneriet committed:
Add report file, folders

Gartneriet committed:
Add report file, folders

Gartneriet committed:
Add report file, folders

tropaadigselv committed:
Wrote license section

Gartneriet committed:
Wrote LLM section

tropaadigselv committed:
Merge branch '118-report-ethics' into 119-license

tropaadigselv committed:
Merge pull request #134 from ITU-BDSA2025-GROUP8/119-license

Wrote license section

CharlottePlanteig committed:
Rephrase some LLM-related sentences.

Gartneriet committed:
Refine language on LLM usage in project

tropaadigselv committed:
Began work in how to run chirp locally

Added instructions for running the Chirp application locally, including prerequisites and commands.

tropaadigselv committed:
Added terminal output for running program

tropaadigselv committed:
Finished run locally

tropaadigselv committed:
small chnage to text

tropaadigselv committed:
Delete docs/images/terminalOutput.png

tropaadigselv committed:
updated image

tropaadigselv committed:
Delete docs/images/terminalOutput.png

tropaadigselv committed:
updated image

tropaadigselv committed:
fixed for new image

Gartneriet committed:
Fix typos and changed language a bit

Gartneriet committed:
Add Visual paradigm project
Add onion architecture sketch

Gartneriet committed:
Onion arch diagram draft
Exported image

Gartneriet committed:
Rename image
Write first draft of report section

Gartneriet committed:
Add todo to self

Gartneriet committed:
Argue entity placement
Update diagram

Gartneriet committed:
rename stuff

Gartneriet committed:
Renamed

Gartneriet committed:
Add pdf export of diagram

Gartneriet committed:
Moved image

Mary249928 committed:
Fix spelling error

Corrected spelling of 'licensed' in the License section.

Mary249928 committed:
Uploaded the domain model diagram #120

Mary249928 committed:
Change image and added vpp project file #120

Mary249928 committed:
Add first draft to Domain Model #120

Gartneriet committed:
Clarify test suite directory

Gartneriet committed:
Fix image path in report documentation

Updated the image path for the codebase structure illustration.

tropaadigselv committed:
Added activity diagrams

tropaadigselv committed:
Document GitHub Actions workflows for Chirp project

Added details about GitHub Actions workflows for building, testing, releasing, and deploying the Chirp application, including activity diagrams.

tropaadigselv committed:
updated image caption

Mary249928 committed:
Reference to diagram and more objective #120

Mary249928 committed:
Capitalize first letter of attributes #120

Gartneriet committed:
Add diagram first draft

Mary249928 committed:
Fix capitalization in domain model section

Corrected capitalization in the domain model description.

Gartneriet committed:
Clarify test package in diagram
Change colors

Gartneriet committed:
Save vpp

Gartneriet committed:
Merge pull request #136 from ITU-BDSA2025-GROUP8/118-report-ethics

118 report ethics

Gartneriet committed:
Slightly modified language

Mary249928 committed:
Add figure text and ! to Chirp #120

Mary249928 committed:
Add formatting

Mary249928 committed:
Change color of diagram

tropaadigselv committed:
updateted diagrams

tropaadigselv committed:
Updated images

CharlottePlanteig committed:
Fix of a couple of typos

CharlottePlanteig committed:
Merge remote-tracking branch 'origin/121-architecture-in-the-small' into 121-architecture-in-the-small

# Conflicts:
#	docs/report.md

CharlottePlanteig committed:
Fix of a couple of typos.

tropaadigselv committed:
Clarify GitHub Actions workflow descriptions in report

Gartneriet committed:
Add private timeline to diagram

Add section text

Gartneriet committed:
Merge pull request #143 from ITU-BDSA2025-GROUP8/121-architecture-in-the-small

121 architecture in the small

tropaadigselv committed:
fixed Chirp to Chirp!

tropaadigselv committed:
fixed Chirp to Chirp!

Gartneriet committed:
First draft of test section

tropaadigselv committed:
Merge pull request #138 from ITU-BDSA2025-GROUP8/127-how-to-make-chirp-work-locally

127 how to make chirp work locally

sarabagger committed:
Inital text about our team work

sarabagger committed:
Rework of sentence structure #126

sarabagger committed:
addition of team work Visual Paradigm files (vpp, png, pdf) #126

sarabagger committed:
Referencing diagram #126

sarabagger committed:
tiny changes, new line etc.

CharlottePlanteig committed:
Add UML Sequence Diagram as png

CharlottePlanteig committed:
Add UML Sequence Diagram as pdf

CharlottePlanteig committed:
Add sequence_of_functionality as a vpp file

CharlottePlanteig committed:
Add report-text to Sequence of functionality calls.

Mary249928 committed:
Merge pull request #149 from ITU-BDSA2025-GROUP8/120-domain-model

120 domain model

Mary249928 committed:
Small fixes: ., ! and break before figure text #148

sarabagger committed:
Additional details in text covering issue analysis & delegation + non-approval of PR #153

sarabagger committed:
Corrected punctuation

sarabagger committed:
Updated team work workflow

Gartneriet committed:
Fix typos

sarabagger committed:
rereview changed to asignee makes changes

Gartneriet committed:
Split diagram into two

Add like and follow to authorized diagram

tropaadigselv committed:
Merge pull request #148 from ITU-BDSA2025-GROUP8/125-build-test-release-and-deployment

125 build test release and deployment

Gartneriet committed:
Playwright installation instructions

Gartneriet committed:
Finish text

CharlottePlanteig committed:
Rephrasing of a sentance

CharlottePlanteig committed:
Merge remote-tracking branch 'origin/124-sequence-of-functionalitycalls-trough-chirp' into 124-sequence-of-functionalitycalls-trough-chirp

Gartneriet committed:
Add alt version with endpoint after logout

Gartneriet committed:
Replace user activities illustration with alt

sarabagger committed:
Merge pull request #154 from ITU-BDSA2025-GROUP8/126-team-work_v2

153 (126) team work

Gartneriet committed:
Merge pull request #152 from ITU-BDSA2025-GROUP8/123-user-activities

123 user activities

sarabagger committed:
Inital text in diagram

sarabagger committed:
Cont. of text on architecture of deployment

tropaadigselv committed:
Revise license section in report.md

Updated license information to include BSD 3-Clause License.

sarabagger committed:
addition of diagram

sarabagger committed:
change of dependency #122

sarabagger committed:
Fix typos in deployment architecture description

Correct typos in the architecture section of the report.

sarabagger committed:
fix of dependency

CharlottePlanteig committed:
Remove old sequence diagram

CharlottePlanteig committed:
Add new sequence diagram

CharlottePlanteig committed:
Add comment about TimelineBaseModel

CharlottePlanteig committed:
Fix typos

Gartneriet committed:
Merge pull request #158 from ITU-BDSA2025-GROUP8/128-how-to-run-test-suite-locally

128 how to run test suite locally

CharlottePlanteig committed:
Merge pull request #156 from ITU-BDSA2025-GROUP8/124-sequence-of-functionalitycalls-through-chirp

124 sequence of functionality calls through chirp

Gartneriet committed:
Fix typo

Corrected spelling errors in the architecture section.

tropaadigselv committed:
Delete docs/diagrams/activity_diagrams.vpp

tropaadigselv committed:
ädded new actibity diagram project file

tropaadigselv committed:
Delete docs/diagrams/build_and_test_activity_diagram.pdf

tropaadigselv committed:
added updated pdf for build and test

tropaadigselv committed:
Delete docs/images/build_and_test_activity_diagram.png

tropaadigselv committed:
added updated build and test png

tropaadigselv committed:
Merge pull request #169 from ITU-BDSA2025-GROUP8/fix-to-build-and-test-activity-diagram

Fix to build and test activity diagram

tropaadigselv committed:
Update report.md

sarabagger committed:
Update report with caption for depl. architecture

Added a note about the deployment architecture of the Chirp! application.

sarabagger committed:
update dependencies

sarabagger committed:
Merge pull request #160 from ITU-BDSA2025-GROUP8/122-architecture-deployed-application

122 architecture of deployed application

sarabagger committed:
Merge branch '115-report' into 116-report-design-and-architecture

sarabagger committed:
Add text about Project Board #126

sarabagger committed:
add picture of project board

Gartneriet committed:
Add <br> between image and caption

Some captions were wrapped incorrectly around the image

Gartneriet committed:
Fix typo

Gartneriet committed:
Move services to infrastructure

Gartneriet committed:
Add <br> to prevent ugly text wrap

Mary249928 committed:
Fix formatting of 'Chirp!' in the main text for consistency

Gartneriet committed:
Rephrase sentence in Architecture of deployed application

Mary249928 committed:
Remove old version of user activities diagram #171

Gartneriet committed:
Merge pull request #171 from ITU-BDSA2025-GROUP8/116-report-design-and-architecture

116 report design and architecture

sarabagger committed:
Merge pull request #173 from ITU-BDSA2025-GROUP8/126-team-work

126 team work

sarabagger committed:
Merge branch '115-report' into 117-report-process

Gartneriet committed:
Fix typo

Mary249928 committed:
Merge pull request #178 from ITU-BDSA2025-GROUP8/117-report-process

117 report process

tropaadigselv commented on docs/report.md:
```diff
@@ -0,0 +1,209 @@
+
+Here comes a description of our domain model.
+
+![Illustration of the _Chirp!_ data model as UML class diagram.](docs/images/domain_model.png)
```
uses wrong image

Mary249928 replied:
Yes it clearly did not overwrite it in the merge

Mary249928 replied:
@Gartneriet fixed it in commit 581c6e2

tropaadigselv commented on docs/report.md:
```diff
@@ -0,0 +1,209 @@
+The first page any Chirp! user sees is the public timeline which displays all cheeps.
+Unauthorized users are limited to browsing cheeps on this public timeline
+and visiting other authors' private timelines. If a user wishes to further interact with
+cheeps and authors, authentication is required as illustrated by the diagram below.
```
should this:

> If a user wishes to further interact with cheeps and authors,

maybe be changed to

> If a user wishes to further interact with the application,

since it is more ambiguous for what the application can do

Mary249928 replied:
I do agree with this - sounds a lot cleaner and is true

sarabagger replied:
good idea

Gartneriet replied:
I agree - it has been changed

tropaadigselv commented on docs/report.md:
```diff
@@ -0,0 +1,209 @@
+
+
+## Team work
+Below is a screenshot of our Project board.
```
maybe some more intro text to why there is a picture of our project board

Mary249928 replied:
I do agree - I just don't have a good line to add....

sarabagger replied:
Is this enough? "The picture below shows the GitHub project board used by the team to coordinate and track development work." It's not really a "why" but at least introductory

Gartneriet replied:
For me, this is a nice to have, but not need to have

tropaadigselv replied:
> Is this enough? "The picture below shows the GitHub project board used by the team to coordinate and track development work." It's not really a "why" but at least introductory

yes that would be great

Gartneriet replied:
That's perfect

tropaadigselv commented on docs/report.md:
```diff
@@ -0,0 +1,209 @@
+<br><br>
+Code generation. Very little code in this project was generated by LLMs and copy/pasted directly,
+and in cases where it was, it is marked by comments and/or in the commit message as co-author.
+The group feels largely unconfortable using AI generated code and opted to only use it in certain cases.
```
unconfortable?

Mary249928 replied:
Changed to uncomfortable in e9519f3

tropaadigselv commented on docs/report.md:
```diff
@@ -0,0 +1,209 @@
+it did not take away from our learning experience but allowed us to prioritize our time and ressources in other areas.
+Additionally, smaller snippits were copy/pasted from ChatGPT when debugging.
+<br><br>
+Overall, the use of LLMs, ChatGPT in particular, has indeed sped up development significantly.
```
sped up?

two e's?

Mary249928 replied:
It can be just the one e or else it would be 'speeded up' - both as far as I understand is correct according to the Cambridge Dictionary. But your attention to detail is lovely!
<img width="909" height="588" alt="image" src="https://github.com/user-attachments/assets/1e9e8fab-d4a3-45ef-b1e2-272077f68a3e" />

Gartneriet replied:
Sped up is the correct term here

tropaadigselv reviewed (changes requested):
image missing and small grammatical changes, see comments for specifics

Gartneriet committed:
Fix domain model section

Mary249928 committed:
Correct spelling errors

Mary249928 commented on docs/report.md:
```diff
@@ -0,0 +1,209 @@
+The code base is structured according to the onion architecture template,
+with one notable exception. Usually, the domain entities are situated within the core of the application,
+which was also the case for this app, before EF Core Identity was implemented.
+With this task came some decisions about how to handle the author/Identity user semantics.
```
Reagrding 'author/Identity user' should it be 'Author/IdentityUser' as above in the Doman Model description?
[If yes, then it should be changed troughout the section]

Gartneriet replied:
Fixed. Author, Cheep and IdentityUser are now consistently referred to as such

Mary249928 reviewed (commented):
I approve this as our final report - It can always get better e.g. see comments - however this version is good enough for me

tropaadigselv committed:
Fix author names in project report

tropaadigselv committed:
Improve link formatting in report.md

Updated links in the documentation for clarity and formatting.

Gartneriet committed:
Fix capitalization and grammar

Corrected capitalization of 'ASP.NET' and improved grammar in the documentation.

Gartneriet committed:
Improve sentence in user activities

sarabagger commented on docs/report.md:
```diff
@@ -0,0 +1,211 @@
+
+### Execute tests
+To run the tests, open a terminal in the project root and run `dotnet test`.
+This should run the entire test suite.
```
This is incredibly nitpicky - but "should" sounds like we aren't sure?

CharlottePlanteig replied:
We can totally change it but I think it is okay. Cambridge dictionary defines it as (B1): "used to show when something is likely or expected". I think that suits it quite well with expected since we expect it to happen but there is a slight possibility og something not working as intended.

Gartneriet replied:
You're right - changed to "...will run the entire..."

Gartneriet committed:
Correct capitalization of Cheep and Author

Gartneriet reviewed (approved):
I am happy with it as is. If a had to point out one thing though, it would be ensuring all images had line breaks `<br>` before and after to prevent funny text wrapping when converting to pdf. But if it is not an issue then it's obviously unnecessary

sarabagger commented on docs/report.md:
```diff
@@ -0,0 +1,211 @@
+<br><br>
+Coder's block. As mentioned, this project introduced many new and unfamiliar concepts,
+which meant that for some tasks, we would have no clue where to begin.
+LLMs were used as a spring board in these cases to get started on a new feature for example.
```
This sentence is weirdly constructed.
What about: "As an example, LLMs were used as a starting point when creating new features"
or "When creating new features, LLMs were consulted to get started"

Gartneriet replied:
Changed to
"As such, LLMs were ocasionally used as a starting point when creating new features or refactoring. For example, ..."
I also feel like it better ties the sentences together.

tropaadigselv committed:
small layout changes and added pdf file

tropaadigselv reviewed (approved):
approved, nice work everyone

CharlottePlanteig commented on docs/report.md:
```diff
@@ -0,0 +1,214 @@
+
+## LLMs, ChatGPT, CoPilot, and others
+ChatGPT and, to a smaller degree, Microsoft Copilot were used during the development of this project.
+Let's briefly discuss how these were used and how helpful they were.
```
"Let's briefly discuss how these were used and how helpful they were." 
I think this is the only place where we use "Let's" which kind of stick out. I don't mind it staying, but potentially to match the tone of the rest of the report it could be something more like " The following describes in which areas of development LLM's where used and how helpful it was" or something like that?

Gartneriet replied:
I agree. It has been changed

CharlottePlanteig reviewed (approved):
Almost done, wuhuuu! So fun to see the final result of our months-long work. I approve! I have made one comment to a phrasing which could be changed but I am okay with it staying as if too. 
Nicely done, everyone!

Gartneriet committed:
Team work introduction

Updated the description of the Project Board screenshot for clarity.

Gartneriet committed:
Fix typo in test execution instructions

Gartneriet committed:
Fix language in LLM section

Gartneriet committed:
Fix language inconsistency in LLM ssection

CharlottePlanteig reviewed (approved):
Looks really good with the new changes!

tropaadigselv committed:
more layout changes

tropaadigselv committed:
llm layout fix
